src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -242,6 +297,140 @@ private static string ValueToHexString(object value)
             return result;
         }
 
+        private static bool InternalFlagsFormat(EnumInfo enumInfo, ulong resultValue, Span<char> destination, out int charsWritten, out bool isDestinationTooSmall)
+        {
+            string[] names = enumInfo.Names;
+            ulong[] values = enumInfo.Values;
+            Debug.Assert(names.Length == values.Length);
+
+            // Values are sorted, so if the incoming value is 0, we can check to see whether
+            // the first entry matches it, in which case we can return its name; otherwise,
+            // we can just return "0".
+            if (resultValue == 0)
+            {
+                if (values.Length > 0 && values[0] == 0)
+                {
+                    string enumName = names[0];
+                    if (destination.Length < enumName.Length)
+                    {
+                        charsWritten = 0;
+                        isDestinationTooSmall = true;
+                        return false;
+                    }
+
+                    enumName.CopyTo(destination);
+                    charsWritten = enumName.Length;
+                    isDestinationTooSmall = false;
+                    return true;
+                }
+
+                destination[0] = '0';
+                charsWritten = 1;
+                isDestinationTooSmall = false;
+                return false;
+            }
+
+            // With a ulong result value, regardless of the enum's base type, the maximum
+            // possible number of consistent name/values we could have is 64, since every
+            // value is made up of one or more bits, and when we see values and incorporate
+            // their names, we effectively switch off those bits.
+            Span<int> foundItems = stackalloc int[64];
```

Comment:
This could be interesting to look into. However, I think it may be out of scope of this PR to keep this focused on simply adding TryFormat without modifying the original Format too much.